web/components/bet/quick-bet.tsx

```diff
@@ -150,57 +154,137 @@ export function QuickBet(props: {
+          onClick={() => placeQuickBet('DOWN')}
+          direction="DOWN"
+          user={user}
+          contract={contract}
+          hover={downHover}
           onMouseEnter={() => setDownHover(true)}
           onMouseLeave={() => setDownHover(false)}
-          onClick={() => placeQuickBet('DOWN')}
-        >
-          <TriangleLeftFillIcon
-            className={clsx(
-              'mx-auto h-6 w-6',
-              downHover ? 'text-indigo-700' : 'text-indigo-500'
-            )}
-          />
-          <span
-            className={clsx(
-              'text-sm font-light text-indigo-500 transition-opacity',
-              downHover ? 'opacity-100' : 'opacity-0 '
-            )}
-          >
-            M$10
-          </span>
-        </Row>
-        <Row
-          className="items-center pl-4"
+        />
+        <BinaryQuickBetButton
+          onClick={() => placeQuickBet('UP')}
+          direction="UP"
+          user={user}
+          contract={contract}
+          hover={upHover}
           onMouseEnter={() => setUpHover(true)}
           onMouseLeave={() => setUpHover(false)}
-          onClick={() => placeQuickBet('UP')}
-        >
-          <span
-            className={clsx(
-              'transition- text-sm font-light text-indigo-500',
-              upHover ? 'opacity-100' : 'opacity-0 '
-            )}
-          >
-            M$10
-          </span>
-          <TriangleRightFillIcon
-            className={clsx(
-              'mx-auto h-6 w-6',
-              upHover ? 'text-indigo-700' : 'text-indigo-500'
-            )}
-          />
-        </Row>
+        />
       </Row>
       <QuickOutcomeView contract={contract} previewProb={previewProb} />
     </div>
   )
 }
 
+function BinaryQuickBetButton(props: {
+  onClick: () => void
+  direction: 'UP' | 'DOWN'
+  user: User
+  contract: Contract
+  hover: boolean
+  onMouseEnter: () => void
+  onMouseLeave: () => void
+}) {
+  const {
+    onClick,
+    direction,
+    user,
+    contract,
+    hover,
+    onMouseEnter,
+    onMouseLeave,
+  } = props
+  const userBets = useUserContractBets(user.id, contract.id)
+  let yesWinnings,
+    noWinnings,
+    position,
+    outcome,
+    invested = null
+  const bets = userBets?.filter((b) => !b.isAnte)
+  if (bets) {
+    const metrics = getContractBetMetrics(contract, bets)
+    invested = metrics.invested
+    const excludeSales = bets.filter((b) => !b.isSold && !b.sale)
+    yesWinnings = sumBy(excludeSales, (bet) =>
+      calculatePayout(contract, bet, 'YES')
+    )
+    noWinnings = sumBy(excludeSales, (bet) =>
+      calculatePayout(contract, bet, 'NO')
+    )
+    position = yesWinnings - noWinnings
+    outcome = position < 0 ? 'NO' : 'YES'
+  }
+  let hasInvestment = false
+  if (direction === 'UP') {
+    hasInvestment = outcome === 'YES' && invested != null && floor(invested) > 0
+  } else {
+    hasInvestment = outcome === 'NO' && invested != null && floor(invested) > 0
+  }
```

Comment:
I think you can get this with:
 `const { invested } = getContractBetMetrics(contract, userBets)`
and
` const { hasYesShares, hasNoShares } = useSaveBinaryShares(contract, userBets)`